### PR TITLE
Update telemetry events for @powerpages 

### PR DESCRIPTION
**Pull request by amitjoshi438**
This pull request includes changes to update telemetry events. The `sendTelemetryEvent` function now includes a new property `userId` in the telemetry data properties. Additionally, the `ControlClassIdMap` now includes a new constant `PowerPagesParticipantDocLink` with a link to a Microsoft documentation page. The `fetchRelatedFiles` and `ECSFeaturesClient` functions now include logging using the `oneDSLoggerWrapper` module. Finally, the `handleChatParticipantFeedback` function now includes logging for thumbs-up and thumbs-down feedback using the `oneDSLoggerWrapper` module.

### CODE REVIEW

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 1]
```diff
@@ -122,11 +127,18 @@ export class PowerPagesChatParticipant {
 
             userPrompt = removeChatVariables(userPrompt);
 
+            this.telemetry.sendTelemetryEvent(VSCODE_EXTENSION_GITHUB_POWER_PAGES_AGENT_SUCCESSFUL_PROMPT, {sessionId: this.powerPagesAgentSessionId, orgId: this.orgID, environmentId: this.environmentID, userId: userId });
```
Comment (gshivi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:130]: What kind of userID being logged here?

File: src/common/copilot/PowerPagesCopilot.ts [hunk 2]
```diff
@@ -416,7 +416,7 @@ export class PowerPagesCopilot implements vscode.WebviewViewProvider {
 
         orgID = activeOrg.OrgId;
         environmentName = activeOrg.FriendlyName;
-        userID = activeOrg.UserId;
+        // userID = activeOrg.UserId;
```
Comment (gshivi) [inline src/common/copilot/PowerPagesCopilot.ts:419]: NIT: Can this be removed?

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts (hunk 1 above)
Comment (amitjoshi438) [reply src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:130]: AAD ID

**Review by gshivi — COMMENTED** [review]
Please validate once that required events are flowing in telemetry clusters.

**Review by gshivi — APPROVED** [review]

File: src/common/utilities/Utils.ts [hunk 3]
```diff
@@ -285,6 +286,7 @@ export async function fetchRelatedFiles(activeFileUri: vscode.Uri, componentType
     } catch (error) {
         const message = (error as Error)?.message;
         telemetry.sendTelemetryErrorEvent(COPILOT_RELATED_FILES_FETCH_FAILED, { error: message, sessionId: sessionId });
+        oneDSLoggerWrapper.getLogger().traceError(COPILOT_RELATED_FILES_FETCH_FAILED, message, error as Error, { sessionId:sessionId }, {});
```
Comment (tyaginidhi) [inline src/common/utilities/Utils.ts:289]: is this a good naming convention for copilot events?

File: src/common/ecs-features/ecsFeatureClient.ts [hunk 4]
```diff
@@ -33,10 +34,12 @@ export abstract class ECSFeaturesClient {
 
             // capture telemetry
             telemetry.sendTelemetryEvent(ECSConfigSuccessfulInit, { clientName: clientName, configFlagCount: Object.keys(this._ecsConfig).length.toString() });
```
Comment (tyaginidhi) [inline src/common/ecs-features/ecsFeatureClient.ts:36]: do we need this anymore then?

File: src/common/ecs-features/ecsFeatureClient.ts [hunk 5]
```diff
@@ -33,10 +34,12 @@ export abstract class ECSFeaturesClient {
...
+            oneDSLoggerWrapper.getLogger().traceInfo(ECSConfigSuccessfulInit, { clientName: clientName, configFlagCount: Object.keys(this._ecsConfig).length.toString() });
         } catch (error) {
             const message = (error as Error)?.message;
             // Log error
             telemetry.sendTelemetryErrorEvent(ECSConfigFailedInit, { error: message });
```
Comment (tyaginidhi) [inline src/common/ecs-features/ecsFeatureClient.ts:41]: same for this event?

File: src/common/chat-participants/powerpages/PowerPagesChatParticipantConstants.ts [hunk 6]
```diff
@@ -49,5 +49,10 @@ export const VSCODE_EXTENSION_GITHUB_POWER_PAGES_AGENT_SCENARIO = 'GitHubPowerPa
 export const VSCODE_EXTENSION_GITHUB_POWER_PAGES_AGENT_SCENARIO_FEEDBACK_THUMBSUP = 'GitHubPowerPagesAgentScenarioFeedbackThumbsUp';
 export const VSCODE_EXTENSION_GITHUB_POWER_PAGES_AGENT_SCENARIO_FEEDBACK_THUMBSDOWN = 'GitHubPowerPagesAgentScenarioFeedbackThumbsDown';
 export const VSCODE_EXTENSION_GITHUB_POWER_PAGES_AGENT_ERROR = 'GitHubPowerPagesAgentError';
+export const VSCODE_EXTENSION_GITHUB_POWER_PAGES_AGENT_WEBPAGE_RELATED_FILES = 'GitHubPowerPagesAgentWebpageRelatedFiles';
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipantConstants.ts:52]: lets discuss once if we have all the telemetry needed

**Review by tyaginidhi — APPROVED** [review]

File: src/common/ecs-features/ecsFeatureClient.ts (hunk 4 above)
Comment (amitjoshi438) [reply src/common/ecs-features/ecsFeatureClient.ts:36]: Yes, for sending to older cluster. As per discussion with Bidisha that is not deprecated yet.

File: src/common/ecs-features/ecsFeatureClient.ts (hunk 5 above)
Comment (amitjoshi438) [reply src/common/ecs-features/ecsFeatureClient.ts:41]: Check the comment above.

File: src/common/utilities/Utils.ts (hunk 3 above)
Comment (amitjoshi438) [reply src/common/utilities/Utils.ts]: updated

**Pull request merged**